### PR TITLE
Omit base64 image fields from cards when sending to matrix (and hence AI bot)

### PR DESCRIPTION
**Pull request by habdelra**
This PR is based on PR https://github.com/cardstack/boxel/pull/1060. Please review and merge that one first.

This PR strips out base64 image fields when sending the cards to matrix, which in turn prevents the aibot from having to deal with this field. OpenAI doesn't support base64 encoding (I asked it, and it confirmed as such), so these fields take up unnecessary space in our prompts to open ai.

### CODE REVIEW

**Review by backspace — APPROVED** [review]
(I changed the base branch to review this so I could see only the relevant changes, will change back after submitting this)

Looks good to me, I wonder if we might want a more general way to mark certain fields as not for serialisation, but that can definitely stay a future consideration.

ETA: I think the Percy diffs would be fixed by merging from `main`.

File: packages/matrix/tests/messages.spec.ts [hunk 1]
```diff
@@ -168,25 +173,70 @@ test.describe('Room messages', () => {
   test('can add a card that is over 65K to a message (i.e. split card into multiple matrix events)', async ({
     page,
   }) => {
-    const testCard = `${testHost}/mango-puppy`; // this is a 153KB card
+    const testCard = `${testHost}/big-card`; // this is a 153KB card
```
Comment (jurgenwerk) [inline packages/matrix/tests/messages.spec.ts:174]: Is the comment still valid for this file?

**Review by jurgenwerk — APPROVED** [review]
👍

**Comment by habdelra** [comment]
> (I changed the base branch to review this so I could see only the relevant changes, will change back after submitting this)
> 
> Looks good to me, I wonder if we might want a more general way to mark certain fields as not for serialisation, but that can definitely stay a future consideration.
> 
> ETA: I think the Percy diffs would be fixed by merging from `main`.

Yeah, that’s why I added an `omitFields` option to our serializeCard api function. It works based on field type

File: packages/matrix/tests/messages.spec.ts (hunk 1 above)
Comment (habdelra) [reply packages/matrix/tests/messages.spec.ts:174]: I’ll update it

Comment (habdelra) [reply packages/matrix/tests/messages.spec.ts:174]: actually this comment is still correct. the big-card uses the same data as the previous mango-puppy card

**Comment by IanCal** [comment]
Note for history - we don't want to pass these straight through, and when it comes to actually using images with a multimodal LLM we'll want to pass them over as a custom field so filtering them here makes sense. Also right now we're using the openai models and there is a multimodal option but that model cannot currently do any function calling or restricted output so isn't usable with our assistant right now.

**Pull request merged**